experimental/packages/otlp-transformer/src/common/internal.ts

```diff
@@ -17,7 +17,11 @@ import type { IAnyValue, IKeyValue } from './types';
 import { Attributes } from '@opentelemetry/api';
 
 export function toAttributes(attributes: Attributes): IKeyValue[] {
-  return Object.keys(attributes).map(key => toKeyValue(key, attributes[key]));
+  const result = [];
+  for (const key in attributes) {
```

Comment:
using `x in y` can have issues with some runtimes, it might still be better to use `Object.keys` and just avoid the map call.

Reply:
In what runtimes? I though this is such a fundamental feature, tbh. The downside is that `Object.keys` will allocate array

Reply:
I don't remember the runtime, as it was around 2 years ago and it was caused (for Application Insights) by the runtime "adding" additional properties to the prototype of Object and Array which when accessed caused and exception. Let me see if I can find the PR and if I added comments for the details.

Reply:
- #1420 Fix issues with for..in usage with prototype extension libraries like ember.js and prototype.js

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties#querying_object_properties
Here what I found in docs, so I used the strictest version [propertyIsEnumerable()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/propertyIsEnumerable)